addons/l10n_my_edi_extended/tests/test_new_submissions_flow.py

```diff
@@ -0,0 +1,657 @@
+            'city': 'Main city',
+            'phone': '+60123456789',
+        })
+        cls.partner_a.write({
+            'vat': 'C2584563201',
+            'l10n_my_identification_type': 'BRN',
+            'l10n_my_identification_number': '202001234568',
+            'country_id': cls.env.ref('base.my').id,
+            'state_id': cls.env.ref('base.state_my_jhr').id,
+            'street': 'that other street, 3',
+            'city': 'Main city',
+            'phone': '+60123456786',
+        })
+        cls.product_a.l10n_my_edi_classification_code = "001"
+
+        # We can reuse this invoice for the flow tests.
+        cls.basic_invoice = cls.init_invoice(
+            'out_invoice', products=cls.product_a
+        )
+        cls.basic_invoice.action_post()
+
+        # For simplicity, we will test everything using a 'test' mode user, but we create it using demo to avoid triggering any api calls.
+        cls.proxy_user = cls.env['account_edi_proxy_client.user']._register_proxy_user(cls.company_data['company'], 'l10n_my_edi', 'demo')
+        cls.proxy_user.edi_mode = 'test'
+
+        cls.env['ir.config_parameter'].set_param('l10n_my_edi.disable.send_and_print.first', 'True')
+
+    @freeze_time('2024-07-15 10:00:00')
+    def test_01_new_basic_submission(self):
+        """
+        This tests the most basic flow: an invoice is successfully sent to the MyInvois platform, and then pass validation.
+        """
+        # Send to MyInvois
+        with patch(CONTACT_PROXY_METHOD, new=self._test_01_mock):
+            self.basic_invoice.action_l10n_my_edi_send_invoice()
+
+        # Now that the invoice has been sent successfully, we assert that some info have been saved correctly.
+        self.assertRecordValues(
+            self.basic_invoice,
+            [{
+                'l10n_my_edi_state': 'valid',
+                'l10n_my_edi_validation_time': datetime.strptime('2024-07-15 05:00:00', '%Y-%m-%d %H:%M:%S'),
+                'l10n_my_edi_invoice_long_id': '123-789-654',
+                'l10n_my_edi_submission_uid': '123456789',
+                'l10n_my_edi_external_uuid': '123458974513518',
+            }]
+        )
+
+        # We will test the actual file in another test class, but we ensure it was generated as expected.
+        self.assertTrue(self.basic_invoice.l10n_my_edi_file_id)
+
+    @freeze_time('2024-07-15 10:00:00')
+    def test_02_new_failed_submission(self):
+        """
+        This test will test a flow where the submission itself (not the documents inside) fails for any reason.
+        A general error as such should be handled, but is not expected and should be treated as a bug on our side.
+
+        As we submit a single invoice, we expect a UserError to be raised.
+        """
+        with patch(CONTACT_PROXY_METHOD, new=self._test_02_mock):
+            with self.assertRaises(UserError, msg='Server error; If the problem persists, please contact the Odoo support.'):
```

Comment:
I believe you wanted to use `assertRaisesRegex`?
The `msg` here is what unittest will show in the logs if the assertion fails